addons/account_edi_ubl_cii/models/account_edi_xml_oioubl_201.py

```diff
@@ -0,0 +1,251 @@
+    'cash': 10,
+    'cheque': 20,
+    'debit': 31,
+    'bank': 42,
+    'card': 48,  # credit card
+    'direct debit': 49,
+    'compensation': 97,
+}
+UBL_TO_OIOUBL_TAX_CATEGORY_ID_MAPPING = {
+    # Simple mapping between tax type provided in UBL and what is accepted in OIOUBL
+    # https://docs.peppol.eu/poacc/billing/3.0/codelist/UNCL5305/
+    # https://www.oioubl.info/codelists/en/urn_oioubl_id_taxcategoryid-1.1.html
+    'AE': 'ReverseCharge',
+    'E': 'ZeroRated',
+    'S': 'StandardRated',
+    'Z': 'ZeroRated',
+    'G': 'ZeroRated',
+    'O': 'ZeroRated',
+    'K': 'ReverseCharge',
+    'L': 'ZeroRated',
+    'M': 'ZeroRated',
+}
+TAX_POSSIBLE_VALUES = set(UBL_TO_OIOUBL_TAX_CATEGORY_ID_MAPPING.values())
+ENDPOINTID_SCHEMEID = {
+    'DK': 'DK:CVR',
+    'AD': 'AD:VAT',
+    'AL': 'AL:VAT',
+    'AT': 'AT:VAT',
+    'BA': 'BA:VAT',
+    'BE': 'BE:VAT',
+    'BG': 'BG:VAT',
+    'CH': 'CH:VAT',
+    'CY': 'CY:VAT',
+    'CZ': 'CZ:VAT',
+    'DE': 'DE:VAT',
+    'EE': 'EE:VAT',
+    'ES': 'ES:VAT',
+    'EU': 'EU:VAT',
+    'FI': 'FI:VAT',
+    'FR': 'FR:SIRET',
+    'GB': 'GB:VAT',
+    'GR': 'GR:VAT',
+    'HR': 'HR:VAT',
+    'HU': 'HU:VAT',
+    'IE': 'IE:VAT',
+    'IT': 'IT:VAT',
+    'LI': 'LI:VAT',
+    'LT': 'LT:VAT',
+    'LU': 'LU:VAT',
+    'LV': 'LV:VAT',
+    'MC': 'MC:VAT',
+    'ME': 'ME:VAT',
+    'MK': 'MK:VAT',
+    'MT': 'MT:VAT',
+    'NL': 'NL:VAT',
+    'NO': 'NO:VAT',
+    'PL': 'PL:VAT',
+    'PT': 'PT:VAT',
+    'RO': 'RO:VAT',
+    'RS': 'RS:VAT',
+    'SE': 'SE:VAT',
+    'SI': 'SI:VAT',
+    'SK': 'SK:VAT',
+    'SM': 'SM:VAT',
+    'TR': 'TR:VAT',
+    'VA': 'VA:VAT',
+}
+
+
+class AccountEdiXmlOIOUBL201(models.AbstractModel):
+    _name = "account.edi.xml.oioubl_201"
+    _inherit = 'account.edi.xml.ubl_20'
+    _description = "OIOUBL 2.01"
+
+    # -------------------------------------------------------------------------
+    # EXPORT
+    # -------------------------------------------------------------------------
+
+    def _export_invoice_filename(self, invoice):
+        return f"{invoice.name.replace('/', '_')}_oioubl_201.xml"
+
+    def _export_invoice_ecosio_schematrons(self):
+        return {
+            'invoice': 'org.oasis-open:invoice:2.0',
+            'credit_note': 'org.oasis-open:creditnote:2.0',
+        }
+
+    def _export_invoice_vals(self, invoice):
+        vals = super()._export_invoice_vals(invoice)
+        vals['vals'].update({
+            'customization_id': 'OIOUBL-2.01',
+            # ProfileID is the property that define which documents the company can send and receive
+            # 'Procurement-BilSim-1.0' is the simplest one: invoice and bill
+            # https://www.oioubl.info/documents/en/en/Guidelines/OIOUBL_GUIDE_PROFILES.pdf
+            'profile_id': 'Procurement-BilSim-1.0',
+            'profile_id_attrs': {
+                'schemeID': 'urn:oioubl:id:profileid-1.6',
+                'schemeAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
```

Comment:
No, it's not. It's an EDI that has been develop by Northern countries (UK, NO, FI, ...)
But you're right in the fact I'm specifically implementing it for DK atm but it could be easily changed to allow other countries later on.
It would just need to put stuff like `agency_map.get(self.env.company.country_id)` or something like that.
I can't do it atm as I lack concrete examples for other countries and I'm short on time.

Reply:
I see in https://github.com/odoo/odoo/blob/master/addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py that we indeed only use a single file for different customizations depending on a country. 
But I see that for Bis3 we use the customer/supplier country_code, aren't there also conditions that we could put for Denmark? (I don't know if the format is supposed to be used when your company is Danish, or if it the partner or something else). 
I know that we won't have values for other countries, but I'm not sure inputting Danish values of UK customer is better

Reply:
Discussed IRL:
- It is now a separate module
- It is probably only going to be used between Danish parties
- Current implementation should support foreigners case too
- We will iterate if required